### ISSUE TITLE
Multi-camera smartphones selecting the regular lens

### ISSUE THREAD

**Issue by cconnor12345** [jeromeetienne/AR.js#619]
**Is your feature request related to a problem? Please describe.**
#581 
#571 

Android phones that have multiple rear cameras tend to use the wrong lens, such as a telescopic lens. In my experience, the Huawei mate 20 pro uses the 3x lens.
iPhones don't have this issue however as they only let you choose between Front or Rear.

**Solution**
From playing around with the code in #571 I have managed to make it so on launch, the correct lens is used on startup.

Here is the code I have altered. From line 6029 in aframe-ar.js, replace the code up until the 
// domElement.play(); on line 6098

```
	// get available devices
	navigator.mediaDevices.enumerateDevices().then(function(devices) {

		var deviceToUse = "";
		var deviceLabel = ""; //Used to check if permissions have been accepted, enumerateDevices() runs without needing to accept permissions
		var isAndroid = false;
		//For android phones
		for(var i = 0; i < devices.length; i++){
			deviceLabel = devices[i].label;
			if(devices[i].kind === "videoinput"){
				if(devices[i].label.includes("0")){
					deviceToUse = devices[i].deviceId;
					isAndroid = true;
					break;
				}
			}
		}
		//For Iphones
		if(isAndroid === false){
			for(var i = 0; i < devices.length; i++){
				if(devices[i].kind === "videoinput"){
					if(devices[i].label.includes("Back")){
						deviceToUse = devices[i].deviceId;
						break;
					}
				}
			}
		}

		var userMediaConstraints;
		if(deviceToUse !== ""){
			userMediaConstraints = {
			audio: false,
				video: {
					deviceId : deviceToUse
				}
			};
		}else{
			userMediaConstraints = {
				audio: false,
				video: {
					facingMode: 'environment',
					width: {
						ideal: _this.parameters.sourceWidth,
						// min: 1024,
						// max: 1920
					},
					height: {
						ideal: _this.parameters.sourceHeight,
						// min: 776,
						// max: 1080
					}
				}
			};
		}
		// get a device which satisfy the constraints
		navigator.mediaDevices.getUserMedia(userMediaConstraints).then(function success(stream) {
			//If permissions have been accepted for the first time, device label will show as ""
			if(deviceLabel === ""){
				location.reload();
			}
			// set the .src of the domElement
            domElement.srcObject = stream;

			var event = new CustomEvent('camera-init', {stream: stream});
			window.dispatchEvent(event);
			// to start the video, when it is possible to start it only on userevent. like in android
			document.body.addEventListener('click', function(){
				domElement.play();
			});
```

I have had to do location.reload(); after navigator.mediaDevices.getUserMedia() as navigator.mediaDevices.enumerateDevices() runs without permissions causing the deviceLabels to not appear, so after you accept permissions the page will refresh meaning navigator.mediaDevices.enumerateDevices() will be able to get device labels.

Forgive some of the sloppy coding but I'm not a JS developer.
This is the solution I will be using for my project as most new android phones now have multiple cameras so when I get people to use my project their camera isn't zoomed in.

**Testing**
I have managed to test on:
Huawei mate 20 pro
iPhone 11
Some Samsung phone with multiple cameras I borrowed but don't know the model.

**Comment by nicolocarpignoli** [comment]
@cconnor12345 thank you so much! can you make a PR for this instead of an issue? thanks :)

**Comment by jamess922** [comment]
Hi ：
    Test with my Huawei mate 20 pro and iPhone 7，the web page was white ，looks the camera didn't working，i use your aframe.ar.js file ,is there something else i need to  change

**Comment by mjbenitez** [comment]
Hi! have you tested it with an iOs device configured in a different language? I have the same issue and still wondering how can I fix it. I attached two screenshots of different iOs devices (iPhone 11 and iPad Air 3) configured in Spanish and German.

Thanks! 

<img width="415" alt="Screenshot 2019-12-18 at 16 33 05" src="https://user-images.githubusercontent.com/17424991/71101153-83f72c80-21b6-11ea-9c4b-941d5b951443.png">
<img width="289" alt="Screenshot 2019-12-17 at 15 58 33" src="https://user-images.githubusercontent.com/17424991/71101186-8ce7fe00-21b6-11ea-9afe-8ffde91adb18.png">

**Comment by eCronik** [comment]
Hey, I've tried integrating this into my aframe-ar.js - but now the website is in an infinite reloading-loop (wanting to accept camera access). Is this fix still valid for v3.3.1?
